ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse.yaml

```diff
@@ -70,7 +70,9 @@ tests:
   interval: 960h
   steps:
     cluster_profile: vsphere
-    workflow: cucushift-installer-rehearse-vsphere-ipi-ovn-shared-to-local-gateway-mode-migration
+    env:
```

Comment:
I think we should move the env setting to ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/ovn/local-to-shared-gateway-mode-migration/provision/cucushift-installer-rehearse-vsphere-ipi-ovn-local-to-shared-gateway-mode-migration-provision-chain.yaml instead of changing the job configuration.

Reply:
@jianlinliu Thanks for looking. Once this merged, I have plans to move that ```env``` under https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml and older releases wherever applicable. Would that work?  Thats what [dev does here](https://github.com/tssurya/release/blob/4d6529be861317df2f21e774c0e4c751733c626b/ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-master.yaml#L54). Let me know

Reply:
It should work (installer team already merged many similar install workflow). But if moving the `env` to chain yaml file, when some other body is trying to integrating e2e testing against the install, he/she only need to reference the provision chain name, no need to copy the env setting in the job configuration again, that will be more tidy, and more easier to collaborate between subteams.

Reply:
Sounds good. Just did that. Fingers crossed :)

Reply:
@jianlinliu The test seems passed though but it seems fishy to me. I expected to see this [logic ](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ovn/local-to-shared-gateway-mode-migration/ovn-local-to-shared-gateway-mode-migration-commands.sh) to be executed in e2e logs but its missing. It was clearly executing when the ```env``` logic was present in ```openshift-verification-tests-master__installer-rehearse.yaml``` . That logic is nothing but ```GATEWAY_MODE: local ```

Reply:
Here is your job log:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/28695/rehearse-28695-periodic-ci-openshift-verification-tests-master-installer-rehearse-installer-rehearse-vsphere/1529085576362332160

```
INFO[2022-05-24T13:47:10Z] Running step installer-rehearse-vsphere-ovn-local-to-shared-gateway-mode-migration. 
INFO[2022-05-24T13:52:50Z] Step installer-rehearse-vsphere-ovn-local-to-shared-gateway-mode-migration succeeded after 5m40s. 
INFO[2022-05-24T13:52:50Z] Step phase pre succeeded after 47m40s.
```

In the job's artifacts, refer to more details in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/28695/rehearse-28695-periodic-ci-openshift-verification-tests-master-installer-rehearse-installer-rehearse-vsphere/1529085576362332160/artifacts/installer-rehearse-vsphere/ovn-local-to-shared-gateway-mode-migration/build-log.txt
```
Changing to shared gateway mode
-------------------
network.operator.openshift.io/cluster patched
clusteroperator.config.openshift.io/network condition met..
<--snip-->
Waiting for daemon set "ovnkube-node" rollout to finish: 5 out of 6 new pods have been updated...
Waiting for daemon set "ovnkube-node" rollout to finish: 5 of 6 updated pods are available...
daemon set "ovnkube-node" successfully rolled out.
<--snip-->
Waiting for daemon set "ovnkube-master" rollout to finish: 2 out of 3 new pods have been updated...
Waiting for daemon set "ovnkube-master" rollout to finish: 2 of 3 updated pods are available...
daemon set "ovnkube-master" successfully rolled out
Routing via host is set to false
Overriding to OVN shared gateway mode was a succes
```

Is that what you are expecting, right? Or do I miss anything?

Reply:
Thanks @jianlinliu . How could I missed investigating artifacts. That logic totally looks functional to me now. Thanks for guidance. Squashed commits as well PTAL